### PR TITLE
Test with Node.js 16.x and fix breaking changes

### LINKED ISSUE #2977: Node 16 Error: re2 fails to build
## Describe the bug
`yarn install` errors in agoric-sdk using Node 16. See below for the log.

## To Reproduce
1. install Node 16
2. on agoric-sdk master, try `yarn install`

## Expected behavior
This should succeed and not error. 

## Platform Environment
 - what OS are you using? MacOS 11.2.3 (20D91)
 - is there anything special/unusual about your platform? Nope
 - what version of the Agoric-SDK are you using? (run `git describe --tags --always`) agoricdev-2-116-g3588c2d8d

## Additional context

```
katesills@kates-mbp agoric-sdk % nvm use 16
Now using node v16.0.0 (npm v7.10.0)
katesills@kates-mbp agoric-sdk % yarn install
yarn install v1.22.5
[1/5] 🔍  Validating package.json...
[2/5] 🔍  Resolving packages...
[3/5] 🚚  Fetching packages...
[4/5] 🔗  Linking dependencies...
warning " > eslint-config-airbnb-base@14.0.0" has incorrect peer dependency "eslint@^5.16.0 || ^6.1.0".
warning " > eslint-plugin-import@2.20.0" has incorrect peer dependency "eslint@2.x - 6.x".
warning " > eslint-plugin-jsx-a11y@6.2.3" has incorrect peer dependency "eslint@^3 || ^4 || ^5 || ^6".
warning " > @agoric/ui-components@0.2.6" has unmet peer dependency "@material-ui/core@^4.11.3".
warning " > @agoric/ui-components@0.2.6" has unmet peer dependency "react@^16.14.0".
warning " > @agoric/ui-components@0.2.6" has unmet peer dependency "react-dom@^16.8.0".
warning "workspace-aggregator-26895d38-0381-4927-86bd-80e11d68f0e0 > @agoric/xsnap > @rollup/plugin-node-resolve@6.1.0" has unmet peer dependency "rollup@^1.20.0".
warning "workspace-aggregator-26895d38-0381-4927-86bd-80e11d68f0e0 > @agoric/xsnap > rollup-plugin-terser@5.3.1" has unmet peer dependency "rollup@>=0.66.0 <3".
warning "workspace-aggregator-26895d38-0381-4927-86bd-80e11d68f0e0 > @agoric/ui-components > eslint-config-react-app@6.0.0" has incorrect peer dependency "eslint-plugin-import@^2.22.0".
warning "workspace-aggregator-26895d38-0381-4927-86bd-80e11d68f0e0 > @agoric/ui-components > eslint-config-react-app@6.0.0" has incorrect peer dependency "eslint-plugin-jsx-a11y@^6.3.1".
[5/5] 🔨  Building fresh packages...
[1/4] ⠂ re2
[-/4] ⠂ waiting...
[3/4] ⠂ node-lmdb
error /Users/katesills/code/agoric-sdk/node_modules/re2: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments: 
Directory: /Users/katesills/code/agoric-sdk/node_modules/re2
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@5.1.0
gyp info using node@16.0.0 | darwin | x64
gyp info find Python using Python version 2.7.16 found at "/System/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python"
(node:45630) [DEP0150] DeprecationWarning: Setting process.config is deprecated. In the future the property will be read-only.
(Use `node --trace-deprecation ...` to show where the warning was created)
gyp info spawn /System/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python
gyp info spawn args [
gyp info spawn args   '/Users/katesills/code/agoric-sdk/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/katesills/code/agoric-sdk/node_modules/re2/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/katesills/code/agoric-sdk/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/katesills/Library/Caches/node-gyp/16.0.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/katesills/Library/Caches/node-gyp/16.0.0',
gyp info spawn args   '-Dnode_gyp_dir=/Users/katesills/code/agoric-sdk/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/katesills/Library/Caches/node-gyp/16.0.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/katesills/code/agoric-sdk/node_modules/re2',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CXX(target) Release/obj.target/re2/lib/addon.o
In file included from ../lib/addon.cc:1:
In file included from ../lib/./wrapped_re2.h:4:
In file included from /Users/katesills/Library/Caches/node-gyp/16.0.0/include/node/node.h:63:
In file included from /Users/katesills/Library/Caches/node-gyp/16.0.0/include/node/v8.h:30:
/Users/katesills/Library/Caches/node-gyp/16.0.0/include/node/v8-internal.h:452:38: error: no template named 'remove_cv_t' in namespace 'std'; did you mean 'remove_cv'?
            !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
                                ~~~~~^~~~~~~~~~~
                                     remove_cv
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/type_traits:697:50: note: 'remove_cv' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_cv
                                                 ^
In file included from ../lib/addon.cc:1:
In file included from ../lib/./wrapped_re2.h:5:
In file included from ../../nan/nan.h:2884:
../../nan/nan_typedarray_contents.h:34:43: warning: 'GetContents' is deprecated: Use GetBackingStore. See http://crbug.com/v8/9908. [-Wdeprecated-declarations]
      data   = static_cast<char*>(buffer->GetContents().Data()) + byte_offset;
                                          ^
/Users/katesills/Library/Caches/node-gyp/16.0.0/include/node/v8.h:5571:3: note: 'GetContents' has been explicitly marked deprecated here
  V8_DEPRECATED("Use GetBackingStore. See http://crbug.com/v8/9908.")
  ^
/Users/katesills/Library/Caches/node-gyp/16.0.0/include/node/v8config.h:415:35: note: expanded from macro 'V8_DEPRECATED'
# define V8_DEPRECATED(message) [[deprecated(message)]]
                                  ^
1 warning and 1 error generated.
make: *** [Release/obj.target/re2/lib/addon.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/katesills/code/agoric-sdk/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (node:events:365:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
gyp ERR! System Darwin 20.3.0
gyp ERR! command "/Users/katesills/.nvm/versions/node/v16.0.0/bin/node" "/Users/katesills/code/agoric-sdk/node_modules/.bin/node-gyp" "rebuild"
gyp ERR! cwd /Users/katesills/code/agoric-sdk/node_modules/re2
gyp ERR! node -v v16.0.0
```

### LINKED ISSUE #3785: `agoric install` fails in demo app on node v16.8.0
Following https://agoric.com/documentation/getting-started/before-using-agoric.html and https://agoric.com/documentation/getting-started/start-a-project.html, I set up a new project with the latest node version installed (v16.8.0). When I do `agoric init demo` then `cd demo`, step 4 fails as shown below. Changing to node v14.15.0 fixed the issue and I was able to successfully run `agoric install`.

```
$ sudo agoric install                                                                       
agoric: install: yarn workspaces --silent info                                                                                          
agoric: install: removing /home/samsiegart/agoric-demo/demo/_agstate/yarn-links                                                          
agoric: install: rm -rf /home/samsiegart/agoric-demo/demo/_agstate/yarn-links                                                            
agoric: install: yarn workspaces --silent info                                                                                          
agoric: install: not linking private package @agoric/swingset-runner                                                                    
agoric: install: not linking private package @agoric/dapp-svelte-wallet-api                                                             
agoric: install: not linking private package @agoric/deployment                                                                          
agoric: install: removing ./node_modules link                                                                                            
agoric: install: removing api/node_modules link                                                                                          
agoric: install: removing contract/node_modules link                                                                                     
agoric: install: removing _agstate/agoric-servers/node_modules link                                                                      
agoric: install: removing ui/node_modules link                                                                                          
agoric: install: updating ./package.json                                                                                                 
agoric: install: updating api/package.json                                                                                               
agoric: install: updating contract/package.json                                                                                          
agoric: install: updating _agstate/agoric-servers/package.json                                                                           
agoric: install: updating ui/package.json                                                                                                
agoric: install: yarn --link-folder=/home/samsiegart/agoric-demo/demo/_agstate/yarn-links install                                        
yarn install v1.22.5                                                                                                                     
$ node -e "process.env.AGORIC_INSTALL && process.exit(0); console.warn('please use: agoric install . For details, see https://agoric.com/documentation/'); process.exit(1)"                                                                                                       
[1/4] Resolving packages...                                                                                                             
[2/4] Fetching packages...                                                                                                               
info fsevents@2.1.3: The platform "linux" is incompatible with this module.                                                              
info "fsevents@2.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.                          
info fsevents@1.2.13: The platform "linux" is incompatible with this module.                                                             
info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.                         [3/4] Linking dependencies...                                                                                                            
warning "workspace-aggregator-b8fc178a-bb2e-4d62-8804-28b516abdc4e > demo-api > @agoric/zoe > @agoric/import-bundle@0.0.11" has unmet peer dependency "ses@^0.10.1".                                                                                                              warning "workspace-aggregator-b8fc178a-bb2e-4d62-8804-28b516abdc4e > demo-agservers > @agoric/cosmic-swingset > @agoric/dapp-svelte-wallet > babel-eslint@11.0.0-beta.2" has unmet peer dependency "@babel/core@>=7.2.0".                                                         [4/4] Building fresh packages...                                                                                                         
[-/6] ⠂ waiting...                                                                                                                       
[2/6] ⠂ re2                                                                                                                              
[3/6] ⠂ node-lmdb                                                                                                                        
[-/6] ⠂ waiting...                                                                                                                       
error /home/samsiegart/agoric-demo/demo/node_modules/node-lmdb: Command failed.                                                          Exit code: 1                                                                                                                             
Command: node-gyp-build                                                                                                                  
Arguments:                                                                                                                               
Directory: /home/samsiegart/agoric-demo/demo/node_modules/node-lmdb                                                                      
Output:                                                                                                                                  
gyp info it worked if it ends with ok                                                                                                    
gyp info using node-gyp@7.1.1                                                                                                            
gyp info using node@16.8.0 | linux | x64                                                                                                 
gyp info find Python using Python version 3.8.10 found at "/usr/bin/python3"                                                             
(node:18576) [DEP0150] DeprecationWarning: Setting process.config is deprecated. In the future the property will be read-only.           (Use `node --trace-deprecation ...` to show where the warning was created)                                                               
gyp info spawn /usr/bin/python3                                                                                                          
gyp info spawn args [                                                                                                                    
gyp info spawn args   '/home/samsiegart/agoric-demo/demo/node_modules/node-gyp/gyp/gyp_main.py',                                         gyp info spawn args   'binding.gyp',                                                                                                     
gyp info spawn args   '-f',                                                                                                             
gyp info spawn args   'make',                                                                                                            
gyp info spawn args   '-I',                                                                                                              
gyp info spawn args   '/home/samsiegart/agoric-demo/demo/node_modules/node-lmdb/build/config.gypi',                                      gyp info spawn args   '-I',                                                                                                              
gyp info spawn args   '/home/samsiegart/agoric-demo/demo/node_modules/node-gyp/addon.gypi',                                              gyp info spawn args   '-I',                                                                                                              
gyp info spawn args   '/root/.cache/node-gyp/16.8.0/include/node/common.gypi',                                                           
gyp info spawn args   '-Dlibrary=shared_library',                                                                                        
gyp info spawn args   '-Dvisibility=default',                                                                                            
gyp info spawn args   '-Dnode_root_dir=/root/.cache/node-gyp/16.8.0',                                                                   
gyp info spawn args   '-Dnode_gyp_dir=/home/samsiegart/agoric-demo/demo/node_modules/node-gyp',                                          gyp info spawn args   '-Dnode_lib_file=/root/.cache/node-gyp/16.8.0/<(target_arch)/node.lib',                                            
gyp info spawn args   '-Dmodule_root_dir=/home/samsiegart/agoric-demo/demo/node_modules/node-lmdb',                                      gyp info spawn args   '-Dnode_engine=v8',                                                                                                
gyp info spawn args   '--depth=.',                                                                                                       
gyp info spawn args   '--no-parallel',                                                                                                   
gyp info spawn args   '--generator-output',                                                                                              
gyp info spawn args   'build',                                                                                                           
gyp info spawn args   '-Goutput_dir=.'                                                                                                   
gyp info spawn args ]                                                                                                                    
gyp info spawn make                                                                                                                      
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]                                                                               
make: Entering directory '/home/samsiegart/agoric-demo/demo/node_modules/node-lmdb/build'                                                  CC(target) Release/obj.target/node-lmdb/dependencies/lmdb/libraries/liblmdb/mdb.o                                                        CC(target) Release/obj.target/node-lmdb/dependencies/lmdb/libraries/liblmdb/midl.o                                                       CXX(target) Release/obj.target/node-lmdb/src/node-lmdb.o                                                                               
In file included from /root/.cache/node-gyp/16.8.0/include/node/v8.h:30,                                                                                  
from ../src/node-lmdb.h:29,                                                                                                              
from ../src/node-lmdb.cpp:24:                                                                                           
/root/.cache/node-gyp/16.8.0/include/node/v8-internal.h: In function ‘void v8::internal::PerformCastCheck(T*)’:                          /root/.cache/node-gyp/16.8.0/include/node/v8-internal.h:488:38: error: ‘remove_cv_t’ is not a member of ‘std’; did you mean ‘remove_cv’?   488 |             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);                                                           |                                      ^~~~~~~~~~~                                                                                       |                                      remove_cv                                                                                   
/root/.cache/node-gyp/16.8.0/include/node/v8-internal.h:488:38: error: ‘remove_cv_t’ is not a member of ‘std’; did you mean ‘remove_cv’?   488 |             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);                                                           |                                      ^~~~~~~~~~~                                                                                       |                                      remove_cv                                                                                   
/root/.cache/node-gyp/16.8.0/include/node/v8-internal.h:488:50: error: template argument 2 is invalid                                      488 |             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);                                                           |                                                  ^                                                                               
/root/.cache/node-gyp/16.8.0/include/node/v8-internal.h:488:63: error: ‘::Perform’ has not been declared                                   488 |             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);                                                           |                                                               ^~~~~~~                                                            
In file included from ../src/node-lmdb.h:30,                                                                                                              
from ../src/node-lmdb.cpp:24:                                                                                           
../src/node-lmdb.cpp: At global scope:                                                                                                   
/root/.cache/node-gyp/16.8.0/include/node/node.h:830:51: warning: cast between incompatible function types from ‘void (*)(v8::Local<v8::Object>)’ to ‘node::addon_context_register_func’ {aka ‘void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, v8::Local<v8::Context>, void*)’} [-Wcast-function-type]                                                                                                                 
830 |       (node::addon_context_register_func) (regfunc),                  \                                                                |                                                   ^                                                                              
/root/.cache/node-gyp/16.8.0/include/node/node.h:848:3: note: in expansion of macro ‘NODE_MODULE_CONTEXT_AWARE_X’                          848 |   NODE_MODULE_CONTEXT_AWARE_X(modname, regfunc, NULL, 0)                                                                               |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                    
../src/node-lmdb.cpp:43:5: note: in expansion of macro ‘NODE_MODULE_CONTEXT_AWARE’                                                          43 |     NODE_MODULE_CONTEXT_AWARE(lmdb, initializeModule)                                                                                  |     ^~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                    
In file included from ../src/node-lmdb.cpp:24:                                                                                           
../src/node-lmdb.h:86:28: warning: ‘lastVersion’ defined but not used [-Wunused-variable]                                                   
86 | static thread_local double lastVersion = 0;                                                                                            |                            ^~~~~~~~~~~                                                                                           
make: *** [node-lmdb.target.mk:136: Release/obj.target/node-lmdb/src/node-lmdb.o] Error 1                                                
make: Leaving directory '/home/samsiegart/agoric-demo/demo/node_modules/node-lmdb/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/samsiegart/agoric-demo/demo/node_modules/node-gyp/lib/build.js:194:23)                  gyp ERR! stack     at ChildProcess.emit (node:events:394:28)           
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)   
gyp ERR! System Linux 5.10.16.3-microsoft-standard-WSL2   
gyp ERR! command "/usr/local/bin/node" "/home/samsiegart/agoric-demo/demo/node_modules/.bin/node-gyp" "rebuild"  
gyp ERR! cwd /home/samsiegart/agoric-demo/demo/node_modules/node-lmdb              
agoric: install: Cannot yarn install 
```

### PR DESCRIPTION
**Pull request by michaelfig**
Fixes #2977
Fixes #3785

Changes needed by Node 16:

* Needs upgrades to `node-lmdb` and `re2` to avoid specifying `-std=c++...` when building extensions
* CI on Ubuntu needs to download `node-canvas` dependencies since there are no prebuilt `linux-node-16` binaries
* `fs.rmdirSync(x, { recursive: true })` is deprecated in favour of `fs.rmSync(x, { recursive: true })`
* Both `rmdirSync` and `rmSync` now throw `ENOENT` if the directory doesn't exist
* `fs.rmSync` needs a `@types/node@^16.0.0` package to pass typechecking
* Zoe test for platform `[...undefined]` error message needs to include Node.js 16: `undefined is not iterable`

### CODE REVIEW

File: packages/swing-store/src/swingStore.js
Comment (dckc) [inline packages/swing-store/src/swingStore.js:122]: they no longer have a "don't bug me if it doesn't exist" option? boring.

**Review by dckc — COMMENTED** [review]
Thanks for taking this on!

I don't think I should approve until I understand better about the unreleased version of node-lmdb.

File: package.json
Comment (michaelfig) [reply package.json:85]: In Yarn, an `"xxx": "xxx"` script entry is already implicitly defined for every `yarn xxx`.  So `yarn patch-package` still works, even though there's no explicitly defined script called `patch-package`.

File: packages/SwingSet/package.json
Comment (michaelfig) [reply packages/SwingSet/package.json]: The unreleased version is in lieu of the new release that I requested at: https://github.com/Venemo/node-lmdb/issues/190#issuecomment-911051769.  Once that release is done, we can upgrade to it.

Indeed the current PR was partly to CI-test the `node-lmdb` fixes that went into that release.

I'm annoyed that `package.json` doesn't allow comments.  That would be useful.

File: packages/zoe/test/unitTests/zcf/test-zoeHelpersWZcf.js
Comment (michaelfig) [reply packages/zoe/test/unitTests/zcf/test-zoeHelpersWZcf.js:267]: Ah, maybe someday we'll have `CODEOWNERS`.

File: package.json
Comment (dckc) [reply package.json:85]: oh... that's a script; I read it as a dependency. gotcha.

**Review by dckc — APPROVED** [review]
:+1:

File: packages/SwingSet/package.json
Comment (michaelfig) [reply packages/SwingSet/package.json]: Venemo/node-lmdb#190 is closed, and `node-lmdb@^0.9.5` is published.  This PR is updated!

File: packages/zoe/test/unitTests/zcf/test-zoeHelpersWZcf.js
Comment (michaelfig) [reply packages/zoe/test/unitTests/zcf/test-zoeHelpersWZcf.js:267]: I'm confident committing this change, but agreed, @katelynsills should take a look at it sometime (in the context of this PR's description) to understand what happened.

Comment (katelynsills) [reply packages/zoe/test/unitTests/zcf/test-zoeHelpersWZcf.js:267]: Looks good!

**Pull request merged**
